### PR TITLE
Fix Metadata ID to conform to NCName

### LINKED ISSUE #3220: Review SAML metadata changes
We noticed that the ID attribute of the SAML metadata changed between two versions of the UAA. 

When setting entityID=`https://localhost:443` in the UAA configuration file (`uaa.yml setting login.entitytID`)
 
UAA 77.19.0

```
ID="https___localhost_443" entityID="https://localhost:443"
```

UAA 77.23.0

```
ID="https://localhost:443" entityID="https://localhost:443"
```

The ID attribute should be a NCName and not contain colons
https://www.datypic.com/sc/xsd/t-xsd_ID.html (type="ID")

I'm opening this issue to track if other changes have occurred, and if we need to be backwards compatible

### PR DESCRIPTION
**Pull request by duanemay**
part of #3220

UAA 77.19.0

`ID="https___localhost_443" entityID="https://localhost:443"
`
UAA 77.23.0

`ID="https://localhost:443" entityID="https://localhost:443"`

### CODE REVIEW

File: server/src/main/java/org/cloudfoundry/identity/uaa/provider/saml/SamlMetadataEntityDescriptorCustomizer.java [hunk 1]
```diff
@@ -89,6 +90,20 @@ public void accept(OpenSamlMetadataResolver.EntityDescriptorParameters entityDes
         }
     }
 
+    private String escapeToNCName(String id) {
+        if (id == null || id.isEmpty()) {
+            id = "_";
```
Comment (strehle) [inline server/src/main/java/org/cloudfoundry/identity/uaa/provider/saml/SamlMetadataEntityDescriptorCustomizer.java:95]: not tested yet

File: server/src/main/java/org/cloudfoundry/identity/uaa/provider/saml/SamlMetadataEntityDescriptorCustomizer.java [hunk 2]
```diff
@@ -89,6 +90,20 @@ public void accept(OpenSamlMetadataResolver.EntityDescriptorParameters entityDes
...
+        }
+
+        // Ensure the ID starts with a letter or underscore
+        if (!Character.isLetter(id.charAt(0)) && id.charAt(0) != '_') {
+            id = "_" + id;
```
Comment (strehle) [inline server/src/main/java/org/cloudfoundry/identity/uaa/provider/saml/SamlMetadataEntityDescriptorCustomizer.java:100]: not tested yet

**Review by strehle — COMMENTED** [review]
please check sonar about test coverage, so either you simplify the logic and then with current tests you cover 80% or create more tests for the edge cases

**Review by strehle — APPROVED** [review]
thanks
